src/edu/washington/escience/myria/storage/Tuple.java

```diff
@@ -30,16 +27,26 @@
   /**
    * The data of the tuple.
    */
-  private final List<Field<?>> data;
+  private final MutableTupleBuffer data;
 
   /**
    * @param schema the schema of the tuple
    */
   public Tuple(final Schema schema) {
+    data = new MutableTupleBuffer(schema);
```

Comment:
What is the rationale for replacing a simple `ArrayList<Field>` with `MutableTupleBuffer`? It appears that `MutableTupleBuffer` carries a lot of overhead for a single-row table, especially when we're creating lots of `Tuple`s (e.g. in your implementation of `Aggregator.addRow()`). Actually, I wonder if we need two versions of `Tuple`: an immutable one which is simply a `ReadableTable` view over selected columns in a selected row of a `ReadableTable`, and a mutable one which implements `AppendableTable` but holds only one row. It seems that combining these two responsibilities in a single class creates unnecessary overhead.